pkg/cloud/cloud.go

```diff
@@ -334,14 +342,20 @@ func newBatcherManager(svc ec2iface.EC2API) *batcherManager {
 		volumeTagBatcher: batcher.New(500, 1*time.Second, func(names []string) (map[string]*ec2.Volume, error) {
 			return execBatchDescribeVolumes(svc, names, volumeTagBatcher)
 		}),
-		instanceIDBatcher: batcher.New(50, 300*time.Millisecond, func(names []string) (map[string]*ec2.Instance, error) {
-			return execBatchDescribeInstances(svc, names)
+		instanceIDBatcher: batcher.New(50, 300*time.Millisecond, func(ids []string) (map[string]*ec2.Instance, error) {
+			return execBatchDescribeInstances(svc, ids)
+		}),
+		snapshotIDBatcher: batcher.New(500, 300*time.Millisecond, func(ids []string) (map[string]*ec2.Snapshot, error) {
```

Comment:
`DescribeSnapshots` can return up to 1000 results before paginating (see `MaxResults` on https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSnapshots.html), instead of the 500 of `DescribeVolumes`, so this should be increased.

Also, why 300ms?

Reply:
> Why 300ms

- DescribeSnapshots will be called at most ~6 times per second (except if the batch size exceeds 1000 in rare cases) 
- Only 300ms worst case latency added to any snapshot RPC. 

As noted in description, a finer tuning of these values should be done once we prioritize snapshot scalability tests.  Starting with a conservative added latency. 

Am open to bumping this up to 500ms if we're worried 6 calls per second is too much.